Lib/logging/__init__.py

```diff
@@ -252,6 +250,20 @@ def _releaseLock():
         _lock.release()
 
 
+@contextmanager
+def _acquireModuleLock():
+    """
```

Comment:
This mthode doesn't acquire the lock. I acquires the lock, hold it during yield, and then release it.

Maybe add a _get_lock() which returns the lock of contextlib.nullmanager, and use: with _get_lock(): ...?

Reply:
Sounds like a better idea, thanks.  I have implemented the change.